relay-server/src/services/processor/span/processing.rs

```diff
@@ -163,6 +181,47 @@ pub fn process(
 
         ItemAction::Keep
     });
+
+    let mut transaction_count = 0;
+    for mut transaction in extracted_transactions {
+        // Give each transaction event a new random ID:
+        transaction.id = EventId::new().into();
+
+        // Enqueue a full processing request for every extracted transaction item.
+        match Envelope::try_from_event(state.envelope().headers().clone(), transaction) {
+            Ok(mut envelope) => {
+                // We don't want to extract spans or span metrics from a transaction extracted from spans,
+                // so set the spans_extracted flag:
+                for item in envelope.items_mut() {
+                    item.set_spans_extracted(true);
+                }
+
+                transaction_count += 1;
+
+                addrs.envelope_processor.send(ProcessEnvelope {
+                    envelope: ManagedEnvelope::standalone(
+                        envelope,
+                        addrs.outcome_aggregator.clone(),
+                        addrs.test_store.clone(),
+                        ProcessingGroup::Transaction,
+                    ),
```

Comment:
I think this should be `ManagedEnvelope::new`, not sure if we should bypass the semaphore here?

Reply:
Good call, I passed in the buffer guard now.